libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1246,7 +1245,7 @@ string ABIFunctions::abiDecodingFunctionByteArray(ArrayType const& _type, bool _
 string ABIFunctions::abiDecodingFunctionCalldataStruct(StructType const& _type)
```

Comment:
I'm confused and don't understand what's the purpose of this function. Why can't `_type.calldataHeadIncrement()` be zero? Why it doesn't need to recurse into the struct members to figure out total minimum length?

Reply:
``calldataHeadIncrement() == 0`` is still the case of "cannot be encoded in calldata" - but as mentioned in the other comments, I think I'll look into changing that to an assertion within ``calldataHeadIncrement`` (and the others) - then we can remove the assertion here.

To avoid overhead we only verify the length on access, that's why we don't recurse - the decoding function will only validate that the currently accessed level is OK, i.e. the heads of all struct members are there - if a dynamically encoded struct member (i.e. a nested struct or nested array) is accessed deeper, we will verify the bounds only during that access.

Reply:
Ah ok. makes sense.